mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,8 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+        require(indices.size == 0 || indices(0) >= 0,
```

Comment:
That's true. 
And if we're adding the one-based check inside the iteration for each indice-value pair, do you think we should check the ascending orders also? 
I thought the original code avoid it intentionally for performance.